modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQJMSContext.java

```diff
@@ -0,0 +1,729 @@
+import javax.jms.TextMessage;
+import javax.jms.Topic;
+
+/**
+ * Implementation of JMSContext for JMS 2.0 Support.
+ * One Context will have only one connection, but can have multiple sessions.
+ */
+public class AMQJMSContext extends Closeable implements AutoCloseable, JMSContext {
+
+    private static final Logger logger = LoggerFactory.getLogger(AMQJMSContext.class);
+
+    /**
+     * The connection for the particular context.
+     */
+    private final AMQConnection connection;
+
+    /**
+     * The acknowledgement mode for the particular context.
+     */
+    private final int sessionMode;
+
+    /**
+     * Multiple sessions possible in one context.
+     */
+    private AMQSession session;
+    private AtomicInteger sessionCount;
+    private static final AtomicInteger initialSessionCount = new AtomicInteger(1);
+
+    /**
+     * To specify whether the underlying connection used by this JMSContext
+     * will be started automatically when a consumer is created.
+     */
+    private boolean autoStart = true;
+
+    /**
+     * Underlying MessageProducer instance shared by all JMSProducers of this JMSContext.
+     */
+    private MessageProducer sharedMessageProducer;
+
+    /**
+     * This is the constructor called when creating the initial JMSContext with a
+     * particular connection.
+     * This is called only from the ConnectionFactory.
+     *
+     * @param connection  underlying connection for this JMSContext
+     * @param sessionMode session Mode for this JMSContext
+     */
+    public AMQJMSContext(AMQConnection connection, int sessionMode) {
+        this(connection, sessionMode, initialSessionCount);
+        if (logger.isDebugEnabled()) {
+            logger.debug("Created JMSContext with Session Mode: {}", sessionMode);
+        }
+    }
+
+    /**
+     * The constructor called with subsequent createContext() calls on the initial JMSContext.
+     *
+     * @param connection   underlying connection used to create initial JMSContext
+     * @param sessionMode  session mode for the new context to be created
+     * @param sessionCount number of sessions/contexts relevant to this connection
+     */
+    private AMQJMSContext(AMQConnection connection, int sessionMode, AtomicInteger sessionCount) {
+        this.connection = connection;
+        this.sessionMode = sessionMode;
+        this.sessionCount = sessionCount;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized JMSContext createContext(int sessionMode) {
+        if (sessionCount.get() != 0) {
+            sessionCount.incrementAndGet();
+            if (logger.isDebugEnabled()) {
+                logger.debug("Created new Context with session mode: {}", sessionMode);
+            }
+            return new AMQJMSContext(connection, sessionMode, sessionCount);
+        } else {
+            throw new JMSRuntimeException("The Connection is Closed.");
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
```

Comment:
While using inheritDoc let's also add some meaningful description on the relevant method